src/audio/google/google_ctc_audio_processing.c

```diff
@@ -372,6 +357,7 @@ static int ctc_prepare(struct processing_module *mod,
 							     /*config=*/NULL,
 							     /*config_size=*/0);
 
+	cd->config = comp_get_data_blob(cd->tuning_handler, NULL, NULL);
```

Comment:
It looks that on `prepare()` CTC is started without any config. So does that mean CTC will always apply config via `reconfigure()`?
As I can figure out, `cd->reconfigure` is toggled up by `ctc_set_config()` only, i.e. a "new" config blob is set explicitly. IIUC, it will not be toggled up in the following cases:
1. CTC gets restarted (`reset()` and then `prepare()`) --> user pauses playback > 10 seconds 
2. config blob is set to CTC while CTS is idle --> depends on start-up sequence\
(reference: https://github.com/thesofproject/sof/blob/main/src/audio/data_blob.c#L538)

I think 2 is easy to reproduce by setting CTC enabled prior to starting playback.

Reply:
Changed to get the config before creating a CTC state then we can pass in a CTC config.